### PR TITLE
Upcoming view implementation

### PR DESCRIPTION
**Pull request by marcosvonstremayr**
This feature consists on creating the upcoming view. For this view I implemented a vertical carousel that has autoplay enabled. This view uses the same BLoC, Use Case, Repository, API Service and Database classes that the Top Rated view uses.

image

https://user-images.githubusercontent.com/109096201/197230397-68f7116c-425e-4d92-a145-9c00c0787535.mp4

### CODE REVIEW

File: lib/presentation/view/upcoming.dart [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        builder: (BuildContext context, AsyncSnapshot<MovieEvent> snapshot) {
+          switch (snapshot.data!.status) {
+            case Status.initial:
+              return const CircularProgressIndicator(
+                valueColor: AlwaysStoppedAnimation<Color>(Palette.primaryLight),
+              );
+            case Status.loading:
+              return const CircularProgressIndicator(
+                valueColor: AlwaysStoppedAnimation<Color>(Palette.primaryLight),
+              );
```
Comment (GARAF004) [inline lib/presentation/view/upcoming.dart]: Instead to return the same, remove one of the returns from a enum call
case Status.initial:
case Status.loading:
   return ...

File: lib/presentation/view/upcoming.dart [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            case Status.success:
+              return CarouselSlider.builder(
+                itemCount: snapshot.data!.movies!.length,
+                itemBuilder: (BuildContext context, int index, int idX) {
```
Comment (GARAF004) [inline lib/presentation/view/upcoming.dart]: trailing comma, use a more descriptive name to replace idX

File: lib/presentation/view/upcoming.dart (hunk 1 above)
Comment (marcosvonstremayr) [reply lib/presentation/view/upcoming.dart]: Done

File: lib/presentation/view/upcoming.dart (hunk 2 above)
Comment (marcosvonstremayr) [reply lib/presentation/view/upcoming.dart]: Done

**Review by LeonardoMolleker — APPROVED** [review]

File: lib/presentation/view/upcoming.dart [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  State<StatefulWidget> createState() => _UpcomingState();
+}
+
+class _UpcomingState extends State<Upcoming> {
+  @override
+  void initState() {
+    Provider.of<IMoviesBloc>(
+      context,
+      listen: false,
+    ).getMovies(ServiceConstants.endpoints['Upcoming']);
```
Comment (MarcosCGlobant) [inline lib/presentation/view/upcoming.dart]: Make 'Upcoming' a constant value

Comment (marcosvonstremayr) [reply lib/presentation/view/upcoming.dart]: Done

**Review by MarcosCGlobant — APPROVED** [review]

**Pull request merged**